### PR TITLE
Add a task base class and basic logger

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**

This PR:
- adds a very basic base class `Task` that will be used when creating new tasks or refactoring existing tasks.
- adds a very basic logger that can write output to stdout as well as a file.
- adds a test for Logger

**Type of change**

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update.  Likely need to add documentation on how to make use of `class Task` and `Logger`.  Will share with relevant developers before making documentation public.

**How Has This Been Tested?**
- Tested in a sandbox.
- Will test in a full featured branch at a later point
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [X] Any dependent changes have been merged and published

### CODE REVIEW

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good to me. Hopefully the logger colors don't add a bunch of junk to output files on certain machines?

**Comment by aerorahul** [comment]
> Looks good to me. Hopefully the logger colors don't add a bunch of junk to output files on certain machines?

It shouldn't since the colored log is only on stdout and not logging to file.
However, if someone pipes stdout to a file, I am not sure.
We can always turn off `colored_log=False` when in production/batch mode.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Could use full docstrings, otherwise looks good.

File: ush/python/pygw/src/pygw/logger.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        logger : Logger object
+        """
+        return self._logger
+
+    @classmethod
+    def add_handlers(cls, logger: logging.Logger, handlers: List[logging.Handler]):
+        """
+        Add a list of handlers to a logger
+
+        Parameters
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygw/src/pygw/logger.py:136]: What should these parameters be? What are their types? (I can get the second by looking at the type hints, but these docstrings will eventually be used by Sphinx or something to create stand-alone documentation webpages.)

File: ush/python/pygw/src/pygw/logger.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+        return logger
+
+    @classmethod
+    def add_stream_handler(cls, level: str = DEFAULT_LEVEL,
+                            _format: str = DEFAULT_FORMAT,
+                            colored_log: bool = False):
+        """
+        Create stream handler
+        This classmethod will allow setting a custom stream handler on children
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygw/src/pygw/logger.py:158]: Parameters? Returns?

File: ush/python/pygw/src/pygw/logger.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+        return handler
+
+    @classmethod
+    def add_file_handler(cls, logfile_path: Union[str, Path],
+                         level: str = DEFAULT_LEVEL,
+                         _format: str = DEFAULT_FORMAT):
+        """
+        Create file handler.
+        This classmethod will allow setting custom file handler on children
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygw/src/pygw/logger.py:191]: Parameters? Returns?

File: ush/python/pygw/src/pygw/task.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        # Store the config and arguments as attributes of the object
+        self.config = config
+
+        for arg in args:
+            setattr(self, str(arg), arg)
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def initialize(self):
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygw/src/pygw/task.py:32]: Short descriptions with what each of these methods should be used for in the child classes?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**